kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/Command.scala

```diff
@@ -0,0 +1,164 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kyuubi.ctl.cmd
+
+import java.net.InetAddress
+
+import org.apache.kyuubi.{KYUUBI_VERSION, KyuubiException, Logging}
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.config.KyuubiConf.{ENGINE_SHARE_LEVEL, ENGINE_SHARE_LEVEL_SUBDOMAIN, ENGINE_TYPE}
+import org.apache.kyuubi.ctl.{CliConfig, ServiceControlObject}
+import org.apache.kyuubi.ctl.ControlCli.printMessage
+import org.apache.kyuubi.ha.HighAvailabilityConf._
+import org.apache.kyuubi.ha.client.{DiscoveryClient, DiscoveryPaths, ServiceNodeInfo}
+
+abstract class Command(var cliArgs: CliConfig) extends Logging {
+
+  val conf = KyuubiConf().loadFileDefaults()
+
+  val verbose = cliArgs.commonOpts.verbose
+
+  def preProcess(): Unit = {
+    this.cliArgs = useDefaultPropertyValueIfMissing()
+    validateArguments()
+  }
+
+  /** Ensure that required fields exists. Call this only once all defaults are loaded. */
+  def validateArguments(): Unit
+
+  def run(): Unit
+
+  def fail(msg: String): Unit = throw new KyuubiException(msg)
+
+  protected def validateZkArguments(): Unit = {
+    if (cliArgs.commonOpts.zkQuorum == null) {
+      fail("Zookeeper quorum is not specified and no default value to load")
+    }
+    if (cliArgs.commonOpts.namespace == null) {
+      fail("Zookeeper namespace is not specified and no default value to load")
+    }
+  }
+
+  protected def validateHostAndPort(): Unit = {
+    if (cliArgs.commonOpts.host == null) {
+      fail("Must specify host for service")
+    }
+    if (cliArgs.commonOpts.port == null) {
+      fail("Must specify port for service")
+    }
+
+    try {
+      InetAddress.getByName(cliArgs.commonOpts.host)
+    } catch {
+      case _: Exception =>
+        fail(s"Unknown host: ${cliArgs.commonOpts.host}")
+    }
+
+    try {
+      if (cliArgs.commonOpts.port.toInt <= 0) {
+        fail(s"Specified port should be a positive number")
+      }
+    } catch {
+      case _: NumberFormatException =>
+        fail(s"Specified port is not a valid integer number: ${cliArgs.commonOpts.port}")
+    }
+  }
+
+  protected def validateUser(): Unit = {
+    if (cliArgs.service == ServiceControlObject.ENGINE && cliArgs.engineOpts.user == null) {
+      fail("Must specify user name for engine, please use -u or --user.")
+    }
+  }
+
+  protected def mergeArgsIntoKyuubiConf(): Unit = {
+    conf.set(HA_ADDRESSES.key, cliArgs.commonOpts.zkQuorum)
+    conf.set(HA_NAMESPACE.key, cliArgs.commonOpts.namespace)
+  }
+
+  private def useDefaultPropertyValueIfMissing(): CliConfig = {
+    var arguments: CliConfig = cliArgs.copy()
+    if (cliArgs.commonOpts.zkQuorum == null) {
+      conf.getOption(HA_ADDRESSES.key).foreach { v =>
```

Comment:
it introduced a regression, assume user upgrading from older version, and use `kyuubi.ha.zookeeper.quorum` instead of `kyuubi.ha.addresses`, the `kyuubi-ctl list` will fail w/ 
```
hadoop@spark5:~/kyuubi_hz_cluster_8/kyuubi$ bin/kyuubi-ctl list
2022-11-30 11:59:45.556 INFO org.apache.kyuubi.Utils: Loading Kyuubi properties from /home/hadoop/kyuubi_hz_cluster_8/kyuubi/conf/kyuubi-defaults.conf
2022-11-30 11:59:45.720 WARN org.apache.hadoop.util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Exception in thread "main" org.apache.kyuubi.KyuubiException: Zookeeper quorum is not specified and no default value to load
	at org.apache.kyuubi.ctl.util.Validator$.fail(Validator.scala:81)
	at org.apache.kyuubi.ctl.util.Validator$.validateZkArguments(Validator.scala:31)
	at org.apache.kyuubi.ctl.cmd.list.ListCommand.validate(ListCommand.scala:27)
	at org.apache.kyuubi.ctl.ControlCliArguments.parse(ControlCliArguments.scala:57)
	at org.apache.kyuubi.ctl.ControlCliArguments.<init>(ControlCliArguments.scala:39)
	at org.apache.kyuubi.ctl.ControlCli$$anon$1$$anon$2.<init>(ControlCli.scala:58)
	at org.apache.kyuubi.ctl.ControlCli$$anon$1.parseArguments(ControlCli.scala:58)
	at org.apache.kyuubi.ctl.ControlCli.doAction(ControlCli.scala:33)
	at org.apache.kyuubi.ctl.ControlCli$$anon$1.doAction(ControlCli.scala:78)
	at org.apache.kyuubi.ctl.ControlCli$.main(ControlCli.scala:86)
	at org.apache.kyuubi.ctl.ControlCli.main(ControlCli.scala)
```

Reply:
ping @lightning-L @turboFei